spec/features/saml/loa1_sso_spec.rb

```diff
@@ -40,6 +40,25 @@
       )
       expect(page).to_not have_css('.accordion-header')
     end
+
+    it 'allows user to view recovery code via profile' do
+      allow(FeatureManagement).to receive(:prefill_otp_codes?).and_return(true)
+      saml_authn_request = auth_request.create(saml_settings)
+      user = create(:user, :with_phone)
+      code = RecoveryCodeGenerator.new(user).create
+
+      visit saml_authn_request
+      sign_in_and_require_viewing_recovery_code(user)
+      click_on(t('shared.nav_auth.my_account'))
+      click_on(t('profile.links.regenerate_recovery_code'))
+
+      expect(current_path).to eq manage_recovery_code_path
+
+      click_acknowledge_recovery_code
+      enter_recovery_code(code: code)
```

Comment:
This helper only works if JS is enabled. It's not raising an error because there are no input fields on the profile page, which is where you end up after `click_acknowledge_recovery_code` when JS is disabled.

By not using JS for this test, it is masking a bug because the modal's form `action` is set to `sign_up_recovery_code_path`, which means the user will still end up at `sign_up/completed`. I verified this twice manually by running the SP rails app locally.

Personally, I think that if a user came from an SP, even if they try to make profile changes during account creation, they should still be able to go back to the SP. Otherwise, they would have to go back to the SP manually and initiate the request again, which they might not know to do.

The behavior where a user goes back to the profile after making profile changes should only apply when the user has visited the site directly IMO.